solr/solrj/src/java/org/apache/solr/client/solrj/impl/HttpSolrClient.java

```diff
@@ -358,7 +358,11 @@ protected HttpRequestBase createMethod(@SuppressWarnings({"rawtypes"})SolrReques
     if (parser == null) {
       parser = this.parser;
     }
-    
+
+    Header[] contextHeaders = new Header[2];
+    contextHeaders[0] = new BasicHeader(CommonParams.SOLR_REQUEST_CONTEXT_PARAM, getContext().toString());
```

Comment:
This seems trivial to forge. Can we tie this into the authentication code paths somehow so that we can really trust that the request is coming from a server instead of just some application claiming to be a server?

Reply:
I am honestly not sure about that -- Can you suggest a mechanism to achieve that? Ideas are more than welcome!

Reply:
https://lucene.apache.org/solr/guide/8_5/authentication-and-authorization-plugins.html#securing-inter-node-requests has some ideas.